Wrappers/Python/ccpi/web_viewer/trame_viewer.py

```diff
@@ -329,50 +325,24 @@ def change_window_level_detail_sliders(self, show_detailed: bool):
         self.slice_window_sliders_are_detailed = show_detailed
 
     def change_slice_window_range(self, window: float, level: float):
-        if hasattr(self.cil_viewer, "setSliceColorWindowLevel"):
-            if self.slice_window_slider_is_percentage:
-                min_percentage = level - window / 2
-                max_percentage = window + level - window / 2
-                self.cil_viewer.setSliceColorPercentiles(min_percentage, max_percentage)
-            else:
-                self.cil_viewer.setSliceColorWindowLevel(window, level)
-        else:
-            self.cil_viewer.setColorWindowLevel(window, level)
-        self.cil_viewer.updatePipeline()
-        self.html_view.update()
+        raise NotImplementedError(
+            "This function is not implemented in the base class, but you can expect an implementation in it's sub"
+            " classes.")
 
     def change_slice_window(self, new_window: float, current_level: float = None):
-        if hasattr(self.cil_viewer, "setSliceColorWindow"):
-            if self.slice_window_slider_is_percentage:
-                if current_level is None:
-                    current_level = new_window
-                self.cil_viewer.setSliceColorPercentiles(new_window, current_level)
-            else:
-                self.cil_viewer.setSliceColorWindow(window=new_window)
-        else:
-            self.cil_viewer.setColorWindow(window=new_window)
-        self.cil_viewer.updatePipeline()
-        self.html_view.update()
+        raise NotImplementedError(
+            "This function is not implemented in the base class, but you can expect an implementation in it's sub"
+            " classes.")
 
     def change_slice_level(self, new_level: float, current_window: float = None):
-        if hasattr(self.cil_viewer, "setSliceColorLevel"):
-            if self.slice_window_slider_is_percentage:
-                if current_window is None:
-                    current_window = new_level
-                self.cil_viewer.setSliceColorPercentiles(current_window, new_level)
-            else:
-                self.cil_viewer.setSliceColorLevel(level=new_level)
-        else:
-            self.cil_viewer.setColorLevel(level=new_level)
-        self.cil_viewer.updatePipeline()
-        self.html_view.update()
+        raise NotImplementedError(
+            "This function is not implemented in the base class, but you can expect an implementation in it's sub"
+            " classes.")
 
     def change_slice_number(self, slice_number):
-        if hasattr(self.cil_viewer, "updateSliceHistogram"):
-            self.cil_viewer.updateSliceHistogram()
-        self.cil_viewer.setActiveSlice(slice_number)
-        self.cil_viewer.updatePipeline()
-        self.html_view.update()
```

Comment:
you could still leave these here, then the 2D viewer would use the parent's change_slice_number unmodified, and the 3D viewer could run self.cil_viewer.updateSliceHistogram() and then call change_slice_number from the parent